### PR TITLE
LPS-121871 Migrate v2 usages in document-library/document-library-web

### PR DESCRIPTION
**Pull request by julien**
The <clay:management-toolbar-v2 /> tag has been replaced with the
<clay:management-toolbar /> tag (which uses clay v3).

**Test plan**:
- Navigate to **Content & Data** > **Documents and Media**
- Uploading new files should work as expected
- Editing tags should work as expected
- Deleting items should work as expected
- Moving items to a new folder should work as expected
- Creating Document Types should work as expected
- Creating Metadata Sets should work as expected

**Test plan to test edit Categories**

- Navigate to Categorization > Categories
- Add New vocabulary
- Add new categories
- Navigate to Content & Data > Documents and Media
- New icon edit categories

**Additional notes**
Originally reviewed [here](https://github.com/liferay-frontend/liferay-portal/pull/805)

**[EDIT]**
In order to test this properly, navigate to **Content & Data** > **Documents and Media**, you'll see three tabs: 
**Documents and Media**, **Document Types** and **Metadata Sets**. 

Each of these tabs has a management toolbar that is affected by these changes, and they should work as they previously used to.

You'll also have to go to the **Documents and Media** tab, create a folder (via the management toolbar), and then click on that folder's name to view and test if the management toolbar works there.

The last thing to do, is to go to the **Document Types**  tab, create a lot of document types (>20) at which point the clicking on the "More" button of the creation menu should open a modal window that has another management toolbar.

Since it seems that it's not possible to create document types right now (changes in DDM), I'm putting this on hold, but feel free to review when DDM is working again.

### CODE REVIEW

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+	return {
+		...otherProps,
+		onActionButtonClick(event, {item}) {
+			const action = item.data?.action;
+
+			if (action === 'checkin') {
+				checkIn();
+			}
+			if (action === 'checkout') {
```
Comment (carloslancha) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:209]: shouldn't this be an `else if`?

Anyway, do we prefer this if-else structure better than a `switch`? (Just asking)

**Review by carloslancha — APPROVED** [review]
Tested and working. LGTM, I just left a minor non-blocking nit. Thx @julien !

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js (hunk 1 above)
Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js:209]: you're right missing an `else` here, good catch. I'll that right now.

Concerning the `switch`, that's a good question, but we've been using this "pattern" so far so I'm just applying the same here.

Comment (julien) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js]: I updated that in [https://github.com/liferay-lima/liferay-portal/pull/1525/commits/82125cf58a4192f968a5906ff3e2cfd5a02aeb07](https://github.com/liferay-lima/liferay-portal/pull/1525/commits/82125cf58a4192f968a5906ff3e2cfd5a02aeb07), thanks for the review @carloslancha (I'm missing the PR is not horrible comments :joy:)

Comment (carloslancha) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js]: This pr is not horrible at all :P

**Comment by carloslancha** [comment]
@boton I'm forwarding this since it's kinda blocking for other prs, feel free to review it and add any comments anyway.

**Comment by carloslancha** [comment]
ci:forward

**Comment by boton** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by boton** [comment]
ci:stop

**Comment by boton** [comment]
Sorry guys I stoped this because @4lejandrito and I are trying to clean as much as possible the old AUI component, avoiding repeat the logic, and forward on TOP of this PR.

What do you think?

**Comment by boton** [comment]
We are testing and edit categories action is missing in the new version, the icon is rendered but not working.

**Comment by carloslancha** [comment]
@boton what AUI is influencing with this pr? I mean, this pr is just replacing the management toolbar, so, why do we need to stop it and block the other prs?

**Comment by carloslancha** [comment]
> We are testing and edit categories action is missing in the new version, the icon is rendered but not working.

Where is this happening?

**Comment by 4lejandrito** [comment]
Hi @carloslancha,

This pull has 2 issues:

1. It introduces a lot of code duplication. Most of the functions inside [main.js](https://github.com/liferay/liferay-portal/blob/72045c9efa696aeb147df30111c0119f4516289a/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/legacy/main.js#L15) are no longer used after these changes. I think we should not duplicate code that can be deleted. 
2. The action to edit categories is not working at all. Is this being implemented in a follow up pull? If so, I think both prs should be merged together and sent as a single one. 

I think we should fix both issues before forwarding this. This will be better in the future if we need to backporting, reverting... etc.

**Comment by boton** [comment]
**Test plan to test edit Categories**

* Navigate to Categorization > Categories
* Add New vocabulary
* Add new categories
* Navigate to Content & Data > Documents and Media
* New icon edit categories

**Comment by julien** [comment]
@boton @4lejandrito @carloslancha, the missing edit categories funcionality has been added in [https://github.com/liferay-lima/liferay-portal/pull/1525/commits/22beb89e07a2928ec08d44bf93e60ab0e8146a60](https://github.com/liferay-lima/liferay-portal/pull/1525/commits/22beb89e07a2928ec08d44bf93e60ab0e8146a60)

**Comment by boton** [comment]
@julien thank you tested and working!

**Comment by boton** [comment]
ci:forward

**Review by boton — APPROVED** [review]

**Comment by Jeremy-Ch** [comment]
ci:forward

**Comment by julien** [comment]
This [task](https://issues.liferay.com/browse/LPS-121871) appeared as closed in Jira so I deleted my branch, but it never got merged in master so resent [here](https://github.com/liferay-lima/liferay-portal/pull/1542)

**Pull request closed without merging**